src/common/cockpitwebserver.c

```diff
@@ -108,6 +110,7 @@ cockpit_web_server_init (CockpitWebServer *server)
                                             cockpit_request_free, NULL);
   server->main_context = g_main_context_ref_thread_default ();
   server->ssl_exception_prefix = g_string_new ("");
+  server->redirect_tls = cockpit_conf_bool ("WebService", "RedirectTLS", TRUE);
```

Comment:
CockpitWebServer is also used in the bridge and elsewhere. I would suggest making this a property, and passing in the setting from src/ws/ specific code.